src/app/History.ts

```diff
@@ -74,4 +74,20 @@ export namespace HistoryManager {
       history.push(history.location.pathname + '?' + urlParams.toString());
     }
   };
+
+  /**
+   * setGraphNamespaceParam: properly replace the namespace section of the url
+   * without going to another page -- just rewrite the namespace part of the url
+   * Also: don't do this if we are on graph/../applications subpage
```

Comment:
The 'applications' conditional is too fragile as proven by the fact that this change doesn't work for any node type other than an app node.  We also have node focus urls for workloads and services (see NavUtils.tsx).  But I'm not recommending adding those to an even bigger conditional, it will still be too fragile.  I recommend changing this to more embody what the comment says it does: "replace the namespace section of the url".   This can't assume the path ends at namespaces/<namespace>.   It needs to replace only the <namespace> part of the path.  Also, if the existing <namespace> == the passed in namespace it doesn't need to do anything.  Note that when drilling into a node the namespace never even changes.